hal_core/src/lib.rs

```diff
@@ -7,27 +7,50 @@ pub enum Error {}
 
 pub type TimerCallbackFn = fn();
 
-pub type Range = (usize, usize);
-// /// A range similar to core::ops::Range but that is copyable.
-// /// The range is half-open, inclusive below, exclusive above, ie. [start; end[
-// #[derive(Debug, Copy, Clone, PartialEq)]
-// pub struct Range<T: Copy> {
-//     pub start: T,
-//     pub end: T,
-// }
-//
-// impl<T: Copy + core::cmp::PartialOrd + core::cmp::PartialEq + core::ops::Sub<Output = T>>
-//     Range<T>
-// {
-//     pub fn new(start: T, end: T) -> Self {
-//         Self { start, end }
-//     }
-//
-//     pub fn contains(&self, val: T) -> bool {
-//         self.start <= val && val < self.end
-//     }
-//
-//     pub fn size(&self) -> T {
-//         self.end - self.start
-//     }
-// }
+/// A range similar to core::ops::Range but that is copyable.
+/// The range is half-open, inclusive below, exclusive above, ie. [start; end[
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct AddressRange {
+    pub start: usize,
+    pub end: usize,
+}
+
+impl AddressRange {
+    pub fn new(start: usize, end: usize) -> Self {
```

Comment:
Could be nice to have a constructor from a `Range<usize>`, the syntax would be a bit more range like: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=38449a783dbfc3192aba75f8c9f6de67

Not sure if that's useful though!